frontend/src/app/shared/form/form.component.html

```diff
@@ -0,0 +1,16 @@
+<form [formGroup]="form">
+  <div class="row" *ngFor="let row of settingsSchema">
+    <ng-container *ngFor="let settings of row">
+      <div *ngIf="shouldDisplayInput(settings)" [class]="settings.class || 'col'">
```

Comment:
maybe I'm missing something but you don't want to display the disabled inputs right?
so maybe instead of calling `shouldDisplayInput` you could use *ngIf="this.form.get(setting.key).disabled" and remove `shouldDisplayInput`